docs/rfcs/0005-migrate-sitepermission-ac-store-geckoview-store.md

```diff
@@ -0,0 +1,37 @@
+---
+layout: page
+title: Migrate feature-sitepermissions to be compatible with the new GeckoView permission API.
+permalink: /rfc/0005-migrate-sitepermission-ac-store-geckoview-store
+---
+
+* Start date: 2020-07-26
+* RFC PR: [#7855](https://github.com/mozilla-mobile/android-components/pull/7855)
+
+## Summary
+
+Migrate our generic the [SitePermissionsStorage](https://github.com/mozilla-mobile/android-components/blob/56e6f5ba55734014804b3f3b0af5c32414fa1d16/components/feature/sitepermissions/src/main/java/mozilla/components/feature/sitepermissions/SitePermissionsStorage.kt#L33) to be compatible with the new GeckoView permissions API.
+
+## Motivation
+
+For permissions, we currently are storing on disk all the site permissions on AC both for GeckoView and WebView, but that's about to change for GeckoView, a new API is about to land, where GeckoView is going to store all the permissions and we are going to have APIs for managing permissions(query, add, delete, and update).
+
+That put us in a situation where we have to decide on how we are going to approach the migration of the existing user data to the new GeckoView APIs.
+
+So far, I think we have to approaches:
+
+1. Keeping both APIs.
+2. Doing a one-shot migration.
+
+#### 1) Keeping both APIs.
+Continue using the current ac API and do a slow migration where every time we query a site permission, we do it in both stores (GeckoView/AC) and when a new site permission is added, we only add it to the new store the geckoView one, this will cause a long tail transition to the new APIs.
+
+#### 2) Doing a one-shot migration.
+We decide a point (when the app startup or other place) to do a background operation to transfer all the data to GeckoView in a one-shot, and until this operation is not finished all the site permissions operations will be put on hold until the migration is over.
```

Comment:
In my case I'm more inclined to go with one time migration as I think I will cause less maintenance work and I think users doesn't have a lot of site permission data so far.

Reply:
Yeah, that sounds reasonable. In this case I think you should describe what you think should happen in the RFC - and you can list other options in the "alternatives" section. Here is the full RFC template (I didn't come around opening the PR for the documentation yet): https://hackmd.io/@n36zdALKTcCzvVhbY3pq5A/rJu3YH3eP

Reply:
Thanks 👍 
I was a bit confused as the intention of this RFC wasn't to have specific approach to be applied, it was more about discussing about which one should we use and then taking this path to be the selected one. Not sure if this a valid use case for RFC?

Reply:
You can start with that (but from reading it I feel like you already have a strong preference), but at the end the RFC should describe something we want to do and agreed on doing. :)

Reply:
👍